src/persist-client/src/internal/state.rs

```diff
@@ -918,6 +918,16 @@ where
             && self.critical_readers.is_empty()
     }
 
+    pub(crate) fn is_empty(&self) -> bool {
```

Comment:
`is_empty` very much sounds to me like "no updates", so I'd like to workshop another name here. Unfortunately, I don't have any counter-proposals for you yet 🙈

Reply:
We discussed this live and decided that it was better to just use something verbose... this API should be rarely used. Going with `is_single_empty_batch` for now but I'm happy to change it if anyone comes up with a better idea!